src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -381,18 +381,42 @@ object RuleUtils {
       filesAppended: Seq[Path]): LogicalPlan = {
     // Transform the location of LogicalRelation with appended files.
     val planForAppended = originalPlan transformDown {
-      case baseRelation @ LogicalRelation(fsRelation: HadoopFsRelation, baseOutput, _, _) =>
-        // Set the same output schema with the index plan to merge them using BucketUnion.
-        val updatedOutput =
-          baseOutput.filter(attr => indexSchema.fieldNames.contains(attr.name))
+      case baseRelation @ LogicalRelation(
+            fsRelation @ HadoopFsRelation(location: PartitioningAwareFileIndex, _, _, _, _, _),
+            baseOutput,
+            _,
+            _) =>
         val newLocation =
-          new InMemoryFileIndex(spark, filesAppended, Map(), None)
-        val newRelation =
-          fsRelation.copy(
-            location = newLocation,
-            dataSchema = StructType(indexSchema.filter(baseRelation.schema.contains(_))),
-            options =
-              fsRelation.options + IndexConstants.INDEX_RELATION_IDENTIFIER)(spark)
+          if (location.partitionSchema.isEmpty) {
+            new InMemoryFileIndex(spark, filesAppended, Map(), None)
+          } else {
+            // Set "basePath" so that partitioned columns are also included in the output schema.
+            val basePath = location.partitionSpec.partitionColumns
+              .foldRight(location.partitionSpec.partitions.head.path) { (col, path) =>
+                if (path.getName.contains(col.name)) {
+                  path.getParent
+                } else {
+                  path
```

Comment:
Is `else` path a possible scenario?

Reply:
Seems it's not possible? null value is handled with default value - `A=a\B=__HIVE_DEFAULT_PARTITION__`